bpfd/src/multiprog/tc.rs

```diff
@@ -134,26 +135,26 @@ impl TcDispatcher {
             Direction::Egress => TcAttachType::Egress,
         };
 
+        let link_id = new_dispatcher.attach_with_options(
+            &iface,
+            attach_type,
+            TcOptions {
+                priority: self.priority,
+                ..Default::default()
+            },
+        )?;
+        let link = new_dispatcher.take_link(link_id)?;
+        self.handle = Some(link.handle());
+        self.link = Some(link);
+
         if let Some(Dispatcher::Tc(mut d)) = old_dispatcher {
-            self.current_pri = d.current_pri - 1;
-            if self.current_pri < MAX_TC_DISPATCHER_PRIORITY {
-                self.current_pri = MIN_TC_DISPATCHER_PRIORITY
-            }
-            let link_id = new_dispatcher.attach(&iface, attach_type, self.current_pri)?;
-            let link = new_dispatcher.take_link(link_id)?;
-            self.link = Some(link);
             // FIXME: TcLinks should detach on drop
```

Comment:
Wait with the new API is this fixed?

Reply:
Not yet.  Drop on detach was fixed by Alessandro pretty recently.  When we go to aya/main it should work and I should be able to remove the explicit detach.